src/general/IWorldEffect.cs

```diff
@@ -36,19 +38,51 @@ public void OnTimePassed(double elapsed, double totalTimePassed)
     {
         var glucose = SimulationParameters.Instance.GetCompound("glucose");
 
+        var totalAmount = 0.0f;
+        var totalChunkAmount = 0.0f;
+
+        var initialDensity = 0.0f;
+        var finalDensity = 0.0f;
+
         foreach (var key in targetWorld.Map.Patches.Keys)
         {
             var patch = targetWorld.Map.Patches[key];
 
+            if (!patch.Biome.Compounds.TryGetValue(glucose, out EnvironmentalCompoundProperties glucoseValue))
+                return;
+
+            totalAmount += glucoseValue.Amount;
+            initialDensity += glucoseValue.Density;
+            totalChunkAmount += patch.GetTotalChunkCompoundAmount(glucose);
+
             // If there are microbes to be eating up the primordial soup, reduce the milk
             if (patch.SpeciesInPatch.Count > 0)
             {
-                if (patch.Biome.Compounds.TryGetValue(glucose, out EnvironmentalCompoundProperties glucoseValue))
-                {
-                    glucoseValue.Density = Math.Max(glucoseValue.Density * Constants.GLUCOSE_REDUCTION_RATE,
-                        Constants.GLUCOSE_MIN);
-                    patch.Biome.Compounds[glucose] = glucoseValue;
-                }
+                glucoseValue.Density = Math.Max(glucoseValue.Density * Constants.GLUCOSE_REDUCTION_RATE,
+                    Constants.GLUCOSE_MIN);
+                patch.Biome.Compounds[glucose] = glucoseValue;
+            }
+
+            finalDensity += patch.Biome.Compounds[glucose].Density;
+        }
+
+        if (finalDensity > 0)
+        {
+            var initialGlucose = Math.Round(initialDensity * totalAmount + totalChunkAmount, 3);
+            var finalGlucose = Math.Round(finalDensity * totalAmount + totalChunkAmount, 3);
+            var percentage = Math.Round((initialGlucose - finalGlucose) / initialGlucose * 100, 1);
+
+            if (percentage >= 20)
+            {
+                targetWorld.LogWorldEvent(new LocalizedString("COMPOUND_CONCENTRATIONS_DRASTICALLY_DROPPED",
```

Comment:
Ah I see you used it here. I very strongly think that this message must be changed to be consistent with the glucose reduction message, as that's the compound that reduces.